### ISSUE TITLE
Synchronous operations are disallowed. Call ReadAsync or set AllowSynchronousIO to true instead

### ISSUE THREAD

**Issue by NongMin** [dotnet/aspnetcore#26920]
#### Version: .Net Core 3.1
```
System.InvalidOperationException: Synchronous operations are disallowed. Call ReadAsync or set AllowSynchronousIO to true instead.
   at Microsoft.AspNetCore.Server.IIS.Core.HttpRequestStream.Read(Byte[] buffer, Int32 offset, Int32 count)
   at Microsoft.AspNetCore.Server.IIS.Core.WrappingStream.Read(Byte[] buffer, Int32 offset, Int32 count)
   at Microsoft.AspNetCore.WebUtilities.HttpRequestStreamReader.ReadIntoBuffer()
   at Microsoft.AspNetCore.WebUtilities.HttpRequestStreamReader.Read(Char[] buffer, Int32 index, Int32 count)
   at System.IO.TextReader.Read(Span`1 buffer)
   at System.IO.TextReader.<>c.<ReadAsyncInternal>b__17_0(Object state)
   at System.Threading.Tasks.Task`1.InnerInvoke()
```

```
streamReader.ReadToEndAsync();
```

```csharp
 public class TextPlainInputFormatter : TextInputFormatter
    {
        public TextPlainInputFormatter()
        {
            SupportedMediaTypes.Add("text/plain");
            SupportedEncodings.Add(UTF8EncodingWithoutBOM);
            SupportedEncodings.Add(UTF16EncodingLittleEndian);
        }

        protected override bool CanReadType(Type type)
        {
            return type == typeof(string);
        }

        public override async Task<InputFormatterResult> ReadRequestBodyAsync(InputFormatterContext context, Encoding encoding)
        {
            string data = null;
            using (var streamReader = context.ReaderFactory(context.HttpContext.Request.Body, encoding))
            {
                data = await streamReader.ReadToEndAsync();
            }
            return InputFormatterResult.Success(data);
        }
    }
```

**Comment by davidfowl** [comment]
Fixed in 5.0 https://github.com/dotnet/aspnetcore/pull/18232. I'd recommend using a normal StreamReader for now if you're trying to do this right now.

**Comment by NongMin** [comment]
Thanks!
